controllers/component_build_controller.go

```diff
@@ -100,124 +101,46 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
-	shouldBuild, err := r.IsNewBuildRequired(ctx, component)
-	if err != nil {
-		return ctrl.Result{}, err
+	if len(component.Annotations) == 0 {
+		component.Annotations = make(map[string]string)
 	}
-	if shouldBuild {
-		if err := r.SubmitNewBuild(ctx, component); err != nil {
-			return ctrl.Result{}, err
-		}
+	if component.Annotations[InitialBuildAnnotationName] == "true" {
+		// Initial build have already happend, nothing to do.
+		return ctrl.Result{}, nil
 	}
 
-	return ctrl.Result{}, nil
-}
-
-var triggerTemplateDiffOpts = cmp.Options{
-	cmpopts.IgnoreFields(triggersapi.TriggerTemplate{}, "TypeMeta", "ObjectMeta"),
-	cmpopts.IgnoreFields(runtime.RawExtension{}, "Raw"),
-	cmp.Comparer(func(x, y resource.Quantity) bool {
-		return x.Cmp(y) == 0
-	}),
-}
-var triggerResourceTemplateDiffOpts = cmp.Options{
-	cmp.Comparer(func(x, y resource.Quantity) bool {
-		return x.Cmp(y) == 0
-	}),
-}
-
-// IsNewBuildRequired detects if a new image should be built for given component.
-// The criterion is equality of existing and expected trigger template of the component.
-func (r *ComponentBuildReconciler) IsNewBuildRequired(ctx context.Context, component appstudiov1alpha1.Component) (bool, error) {
-	log := r.Log.WithValues("Namespace", component.Namespace, "Application", component.Spec.Application, "Component", component.Name)
-
-	expectedTriggerTemplate, err := gitops.GenerateTriggerTemplate(component)
-	if err != nil {
-		return false, err
+	// Set initial build annotation to prevent next builds
+	component.Annotations[InitialBuildAnnotationName] = "true"
+	if err := r.Client.Update(ctx, &component); err != nil {
```

Comment:
On a more philosophical note, I'm wondering if we should really update the Component CR here (even if it's only an annotation). I understood that the idea was to keep all the Component updates only in HAS. 

Are we opening up an exception for annotations?